src/util/secrets/secrets.c

```diff
@@ -1057,15 +1061,29 @@ errno_t sss_sec_get(TALLOC_CTX *mem_ctx,
 
     if (attr_enctype) {
         enctype = sss_sec_str_to_enctype(attr_enctype);
-        ret = local_decrypt(req->sctx, mem_ctx, attr_secret, enctype, _secret);
+        ret = local_decrypt(req->sctx, tmp_ctx, attr_secret, enctype, &secret);
         if (ret) goto done;
     } else {
-        *_secret = talloc_strdup(mem_ctx, attr_secret);
-        if (*_secret == NULL) {
+        secret = talloc_strdup(tmp_ctx, attr_secret);
+        if (secret == NULL) {
             ret = ENOMEM;
             goto done;
         }
     }
+
+    attr_datatype = ldb_msg_find_attr_as_string(res->msgs[0], "type", "simple");
```

Comment:
I would propose to enclose this lookup in `if (_datatype != NULL)`

Reply:
Done.